packages/util/src/exp.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+interface CompatService<T> {
```

Comment:
Can we enforce it by making `_FirebaseService` extend it and have all compat services implement `_FirebaseService` (Some compat services (e.g. rc) are already doing it )? https://github.com/firebase/firebase-js-sdk/blob/master/packages-exp/app-compat/src/types.ts#L37

Reply:
This will require changing a number of other packages so should be done in a separate PR.